azure/scope/clients.go

```diff
@@ -86,70 +92,85 @@ func (c *AzureClients) setCredentialsWithProvider(ctx context.Context, subscript
 		return fmt.Errorf("credentials provider cannot have an empty value")
 	}
 
-	settings, err := c.getSettingsFromEnvironment(environmentName)
+	err := c.getSettingsFromEnvironment(environmentName)
 	if err != nil {
 		return err
 	}
 
 	if subscriptionID == "" {
-		subscriptionID = settings.GetSubscriptionID()
+		subscriptionID = c.SubscriptionID()
 		if subscriptionID == "" {
 			return fmt.Errorf("error creating azure services. subscriptionID is not set in cluster or AZURE_SUBSCRIPTION_ID env var")
 		}
 	}
 
-	c.EnvironmentSettings = settings
-	c.ResourceManagerEndpoint = settings.Environment.ResourceManagerEndpoint
-	c.ResourceManagerVMDNSSuffix = settings.Environment.ResourceManagerVMDNSSuffix
-	c.Values["AZURE_SUBSCRIPTION_ID"] = strings.TrimSuffix(subscriptionID, "\n")
-	c.Values["AZURE_TENANT_ID"] = strings.TrimSuffix(credentialsProvider.GetTenantID(), "\n")
-	c.Values["AZURE_CLIENT_ID"] = strings.TrimSuffix(credentialsProvider.GetClientID(), "\n")
+	c.subscriptionID = strings.TrimSuffix(subscriptionID, "\n")
+	c.tenantID = strings.TrimSuffix(credentialsProvider.GetTenantID(), "\n")
+	c.clientID = strings.TrimSuffix(credentialsProvider.GetClientID(), "\n")
 
 	clientSecret, err := credentialsProvider.GetClientSecret(ctx)
 	if err != nil {
 		return err
 	}
-	c.Values["AZURE_CLIENT_SECRET"] = strings.TrimSuffix(clientSecret, "\n")
+	c.clientSecret = strings.TrimSuffix(clientSecret, "\n")
 
 	c.authType = credentialsProvider.Type()
 
-	tokenCredential, err := credentialsProvider.GetTokenCredential(ctx, c.ResourceManagerEndpoint, c.Environment.ActiveDirectoryEndpoint, c.Environment.TokenAudience)
+	tokenCredential, err := credentialsProvider.GetTokenCredential(ctx, c.ResourceManagerEndpoint, c.activeDirectoryEndpoint, c.tokenAudience)
 	if err != nil {
 		return err
 	}
 	c.TokenCredential = tokenCredential
 	return err
 }
 
-func (c *AzureClients) getSettingsFromEnvironment(environmentName string) (s auth.EnvironmentSettings, err error) {
-	s = auth.EnvironmentSettings{
-		Values: map[string]string{},
-	}
-	s.Values["AZURE_ENVIRONMENT"] = environmentName
-	setValue(s, "AZURE_SUBSCRIPTION_ID")
-	setValue(s, "AZURE_TENANT_ID")
-	setValue(s, "AZURE_AUXILIARY_TENANT_IDS")
-	setValue(s, "AZURE_CLIENT_ID")
-	setValue(s, "AZURE_CLIENT_SECRET")
-	setValue(s, "AZURE_CERTIFICATE_PATH")
-	setValue(s, "AZURE_CERTIFICATE_PASSWORD")
-	setValue(s, "AZURE_USERNAME")
-	setValue(s, "AZURE_PASSWORD")
-	setValue(s, "AZURE_AD_RESOURCE")
```

Comment:
AFAICT none of these environment variables mean anything to the SDKv2 clients that we use.